sql/rowexec/ddl.go

```diff
@@ -1184,9 +1184,6 @@ func (b *BaseBuilder) buildAlterTableCollation(ctx *sql.Context, n *plan.AlterTa
 }
 
 func (b *BaseBuilder) buildCreateForeignKey(ctx *sql.Context, n *plan.CreateForeignKey, row sql.Row) (sql.RowIter, error) {
-	if n.FkDef.OnUpdate == sql.ForeignKeyReferentialAction_SetDefault || n.FkDef.OnDelete == sql.ForeignKeyReferentialAction_SetDefault {
```

Comment:
can you switch on the `b.Parser.(type)` type to either error or resolve the defaults here? that would keep the defaults lifecycle in binding, stay consistent with trying to get rid our analyzer rules

Reply:
I don't see a way to get to the parser from the `rowexec` BaseBuilder, but I like that idea for the `planbuilder` Builder, which I assume is what you're suggesting – we could do the validation check there and avoid having to add the new, specific analyzer validation rule for `SET DEFAULT` (and then not have to remove it in Doltgres, too). That would tidy things up a bit. 

I don't see that we have enough info to resolve the column default value expressions in `planbuilder` yet, since all that info hasn't been added to the plan node yet, but let me know if you see something there and I'm happy to try it out.

Reply:
Sorry I misread the package name, I forgot foreign keys were a holdout that we resolve so late. We need to fix that but maybe inappropriate for this PR.